native/index.d.ts

```diff
@@ -0,0 +1,68 @@
+// <reference path="../react/react-native.d.ts" />
+
+declare module "styled-components/native" {
+    import * as ReactNative from "react-native";
+    import { StatelessComponent } from "react";
+
+    interface StyledFunction<T> {
+      (values: TemplateStringsArray): T;
+      (...values: any[]): T;
```

Comment:
@DanielRosenwasser I just changed the signature to have also the PropTypes, but this won't work with the examples provided, like:

![](http://s.patrick.wtf/asdTlyct6G/Screen-Shot-2016-11-27-09-10-47.png)

Do you think that's fine for an end user? I'm not sure if there could be a way to pass a custom PropTypeClass without creating a new component